### PR TITLE
feat(styles): introduce message popover component [ci visual]

### PR DESCRIPTION
**Pull request by N1XUS**
## Related Issue
Relates https://github.com/SAP/fundamental-ngx/issues/8329

## Description
Added new components:
- Message List
- Message View
- Message Popover

image
image
image
image

#### Please check whether the PR fulfills the following requirements

1. The output matches the design specs
- [x] All values are in `rem`
- [x] Text elements follow the truncation rules
- [x] hover state of the element follow design spec
- [x] focus state of the element follow design spec
- [x] active state of the element follow design spec
- [x] selected state of the element follow design spec
- [x] selected hover state of the element follow design spec
- [x] pressed state of the element follow design spec
- [x] Responsiveness rules - the component has modifier classes for all breakpoints
- [x] Includes Compact/Cosy/Tablet design
- [x] RTL support
2. The code follows fundamental-styles code standards and style
- [x] only one top level `fd-*` class is used in the file
- [x] BEM naming convention is used
- [x] Mixins are used for repeatable code (`fd-rtl`, `fd-ellipsis`, `fd-flex`, `fd-selected`, `fd-focus`, ect.)
- [x] A11y support - keyboard support, screenreader support, proper ARIA attributes, etc.
- [x] `fd-reset()` mixin is applied to all elements
- [x] Variables are used, if some value is used more than twice.
- [x] Checked if current components can be reused, instead of having new code.
3. Testing
- [x] tested Storybook examples with "CSS Resources" `normalize` option 
- [x] tested Storybook examples with "CSS Resources" `unnormalize` option 
- [x] Updated tests
- [x] last commit message should have `[ci visual]` so it can trigger chromatic visual regression (e.g. `test: run chromatic visual regression [ci visual]`)
4. Documentation
- [x] Storybook documentation has been created/updated
- [x] Breaking Changes [wiki](https://github.com/SAP/fundamental-styles/wiki/Breaking-Changes) has been updated in case of breaking changes.

### CODE REVIEW

File: src/styles/mixins/list/_list-message-view.scss [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    .#{$block}__subtitle {
+      padding-top: 0.25rem;
+      font-family: var(--sapFontFamily);
+      font-size: var(--sapFontSize);
+      color: var(--sapContent_LabelColor);
+    }
+
+    .#{$block}__link {
+      &:active {
```
Comment (platon-rov) [inline src/styles/mixins/list/_list-message-view.scss:35]: `fd-active` mixin

**Review by platon-rov — CHANGES_REQUESTED** [review]
Popover should a bit overlap its trigger in Fiori 3
image

Read more should be under the next, not under the icon
image

In details mode we should see back button in header
Let's add example of message popover in details mode so we can track its visuals in chromatic
image

Header elements sticked to the border in HC themes
image

Icons should be filled in Fiori HC themes, not outlined
image

**Comment by N1XUS** [comment]
> Popover should a bit overlap its trigger in Fiori 3 Curr: image Specs: image
> 
> Read more should be under the next, not under the icon Curr: image Specs: image
> 
> In details mode we should see back button in header Let's add example of message popover in details mode so we can track its visuals in chromatic Specs: image
> 
> Header elements sticked to the border in HC themes image
> 
> Icons should be filled in Fiori themes, not outlined Curr: image Specs: image

Hey Platon. Thank you for your review. Couple of notes from my side:
- Filled icons - It's a general issue with the Object Status for Fiori theme. We had an issue for that somewhere, so it's not relevant for this particular task since this component is a combination of existing components with small style additions
- Overlapping popover: UI5's version doesn't overlap, and, frankly, I think such overlapping might be harmful in ngx

**Review by platon-rov — APPROVED** [review]

File: src/styles/message-popover.scss [hunk 2]
```diff
@@ -0,0 +1,103 @@
+@import "./mixins";
+@import "./message-view.scss";
+
+@mixin _fd-message-popover-trigger($colors, $state) {
```
[the remark this answers is not in the document]
Comment (N1XUS) [reply src/styles/message-popover.scss:4]: @InnaAtanasova following button mixins that are limited to the button component only

File: src/styles/_message-view.scss [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  }
+
+  &__details {
+    display: none;
+    position: absolute;
+    top: 0;
+    left: 0;
+    right: 0;
+    bottom: 0;
+    padding: 1rem 1rem 1rem 3.5rem;
```
[the remark this answers is not in the document]
Comment (InnaAtanasova) [reply src/styles/_message-view.scss:23]: Screen Shot 2022-10-20 at 4 21 46 PM

**Comment by InnaAtanasova** [comment]
Missing truncation rules for the title and subtitle.
Fund-styles:
Screen Shot 2022-10-20 at 4 39 51 PM

UI5:
Screen Shot 2022-10-20 at 4 40 11 PM

File: src/styles/_message-view.scss [hunk 4]
```diff
@@ -0,0 +1,55 @@
+@import "./mixins";
+
+$block: #{$fd-namespace}-message-view;
+
+.#{$block} {
+  @include fd-reset();
```
[the remark this answers is not in the document]
Comment (N1XUS) [reply src/styles/_message-view.scss:6]: Scrollbar aren't overwriting necessary parts of the message view. Technically, message view can be also used without fd-scrollbar, thus it also should contain own reset

**Review by InnaAtanasova — APPROVED** [review]

**Comment by InnaAtanasova** [comment]
Great job, Denis!

**Pull request merged**